torch/_dynamo/graph_utils.py

```diff
@@ -1,32 +1,33 @@
 from collections import deque
 from typing import Any
 
-from torch.fx import Graph, Node
-from torch.utils._pytree import tree_flatten
+from torch.fx import Graph, map_arg, Node
+from torch.utils._ordered_set import OrderedSet
 
 
 # flattens with support for slices
 # Note: a better way to do this would
 # be register/unregister slices as pytree nodes
 # but there is no unregister API in the pytorch
 # pytree impl
-def _flatten_args_kwargs(args: Any) -> list[Node]:
-    fully_flattened = []
-
-    def flatten(args: Any) -> None:
-        flattened, _ = tree_flatten(args)
-        for arg in flattened:
-            if isinstance(arg, slice):
-                start = arg.start
-                stop = arg.stop
-                step = arg.step
-                flatten((start, stop, step))
-            else:
-                fully_flattened.append(arg)
-
-    flatten(args)
-
-    return fully_flattened
+def _get_flat_args(
+    node: Node, node_to_additional_deps: dict[Node, OrderedSet[Node]]
+) -> list[Node]:
+    args = list[Any]()
```

Comment:
Why isn't it list of Node?

Reply:
It can be ints, strings, etc.

Reply:
But the return type of the function is list[Node]?

Reply:
good catch, I'll change, map_arg filters ndoes out. Curious why lint didn't catch this